src/Microsoft.Azure.Relay/Common/InputQueue.cs

```diff
@@ -173,7 +166,7 @@ public void Dispatch()
 
             if (outstandingReaders != null)
             {
-                ActionItem.Schedule(s => CompleteOutstandingReadersCallback(s), outstandingReaders);
+                ActionItem.Schedule(CompleteOutstandingReadersCallback, outstandingReaders);
```

Comment:
Call group method instead of lambda calling the method

Reply:
Your suggested version allocates every time the line runs. The existing version allocates and caches a delegate only the version first time. Subsequent calls do no allocate an additional delegates.

Reply:
Ok I wasn't aware of that, will investigate and remove it. thks @dlstucki

Reply:
I made a sample to show what I'm talking about:
``` C#
using System;
using System.Diagnostics;

class Program
{
    static void Main(string[] args)
    {
        const long Iterations = 2 * 1000 * 1000 * 1000L;
        var program = new Program();
        var stopwatch = new Stopwatch();
        stopwatch.Reset();
        GC.KeepAlive(stopwatch.Elapsed);

        Console.WriteLine("Example1");
        stopwatch.Restart();
        for (long i = 0; i < Iterations; i++)
        {
            ActionItemSchedule(s => CompleteWaitersCallback(s), program);
        }
        Console.WriteLine($"Example1 completed in {stopwatch.Elapsed}");

        Console.WriteLine("Example2");
        stopwatch.Restart();
        for (long i = 0; i < Iterations; i++)
        {
            ActionItemSchedule(CompleteWaitersCallback, program);
        }
        Console.WriteLine($"Example2 completed in {stopwatch.Elapsed}");
    }

    static void ActionItemSchedule(Action<object> action, object state)
    {
        action(state);
    }

    static void CompleteWaitersCallback(object state)
    {
        GC.KeepAlive(state);
    }
}
```
Sample output on my machine:
> Example1
> Example1 completed in 00:00:09.7257016
> Example2
> Example2 completed in 00:00:25.8970678

Also, the perf counters show a big difference:
![image](https://user-images.githubusercontent.com/11354387/48499599-a8334d80-e7ed-11e8-9022-3500fc9aea15.png)